src/block_sizes.jl

```diff
@@ -69,3 +65,12 @@ function block_sizes(::Type{T}) where {T}
 
     Mc, Kc, Nc
 end
+
+function _calculate_L3(_L2, L2c, _L3, L3c, st)
+    if (2 * _L2 * L2c) > (_L3 * L3c)
```

Comment:
Use
```julia
if VectorizationBase.CACHE_INCLUSIVITY[3]
    L3 = (StaticInt{_L3}() - StaticInt{_L2}()) ÷ st
else
    L3 = StaticInt{_L3}() ÷ st
end
```

Reply:
> Refactoring `block_sizes` relies on constant propagation for type stability.

Oof. I'd rather not risk it. How about:
1. For now, we revert the refactoring of `block_sizes`.
2. In order to get code coverage in both paths of this `if` statement, we work on implementing an environment variable override feature in `VectorizationBase.jl`?

Can you undo the `block_size` changes and push to this branch? If not, I can do it when I'm back at a computer.

Reply:
Note that on my local master, where I made the switch to use `VectorizationBase.CACHE_INCLUSIVITY[3]` as the check:
```julia
julia> @code_typed Octavian.block_sizes(Float64)
CodeInfo(
1 ─     nothing::Nothing
│       nothing::Nothing
└──     goto #3 if not false
2 ─     nothing::Nothing
3 ┄     nothing::Nothing
└──     return (Static(96), Static(1173), Static(954))
) => Tuple{StaticInt{96}, StaticInt{1173}, StaticInt{954}}

julia> foo() = Octavian.block_sizes(Float64)
foo (generic function with 1 method)

julia> @code_typed foo()
CodeInfo(
1 ─     return (Static(96), Static(1173), Static(954))
) => Tuple{StaticInt{96}, StaticInt{1173}, StaticInt{954}}```
```
`blcok_sizes` are entirely calculated at compile time.

Reply:
Something else you could do is call `_calculate_L3` with `StaticInt`s.

Reply:
Yeah I think it's safer to leave everything in one function (like it is on master). That way, the compiler has the most ability to do optimizations.

Reply:
My suggested fix in the other comment should work perfectly well.

Reply:
Just tested locally:
```julia
julia> @code_typed Octavian.block_sizes(Float64)
CodeInfo(
1 ─     nothing::Nothing
│       nothing::Nothing
│       nothing::Nothing
└──     return (Static(96), Static(1173), Static(954))
) => Tuple{StaticInt{96}, StaticInt{1173}, StaticInt{954}}
```

Reply:
Mostly, I don't trust constant propagation to work reliably and not cause type inference problems.
This is the point of `StaticInt`s: they're compile-time constants.

So as long as creating one can be inferred, i.e. because the number is constant at the location you're define the `StaticInt`, you can pass them around without issue and have things work/be compile time constants.